File: Top Interview 150/Binary Tree General/0129. Sum Root to Leaf Numbers/129. Sum Root to Leaf Numbers.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def sumNumbers(self, root):
        # total = 0
        s = ""
        
        def helper( root, s):
            
            if not root:
                return 0
            
            s += str(root.val)
            
            if not root.left and not root.right:
                return (int(s))
            
            return helper(root.left, s) + helper(root.right,s)
        
        return helper(root,s)

        

File: Top Interview 150/Binary Tree General/0129. Sum Root to Leaf Numbers/129. Sum Root to Leaf Numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumNumbers(TreeNode root) {
        return (findSum(root, 0));
    }
    public int findSum(TreeNode root,int num){
        if(root == null)
            return 0;
        num = (num*10) + root.val;
        // System.out.println(num+","+root.val);
        if(root.right == null && root.left ==null)
            return num;
        return (findSum(root.right, num) + findSum(root.left, num));
    }
}